### PR TITLE
API Access Info endpoint

### LINKED ISSUE #34844: API Access Info endpoint
### Specs:
A default scope has to be defined, otherwise the API access will get access to all the scopes. 

is_authenticated to /api-access/infos → get the info of the scope allowed and the shop associated (the name of the endpoint still have to be defined) 
  - List of shops (no multishop for now)
  - List scopes 
  - All API access info except the secret

--
A query get access info, handle the info of the API access (get the id or to be managed by the handler)
The endpoint should be defined in the core (not in the module)

### Acceptance test
When the endpoint is called with the authorization Then
  - List of shops associated to the API access
  - List scopes allowed
  - All API access info except the secret are displayed
  - Add integration tests in the core

### Technical note
First we should check if the `is_authenticated` scope is really useful to access public endpoints (without any scopes assigned on it), and if the other ones are correctly protected If yes then we could remove the `is_authenticated` scope completely

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | API Access Info endpoint
| Type?             | improvement
| Category?         | BO 
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI / tests UI are green
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/8022234004
| Fixed issue or discussion?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/34844
| Related PRs       | https://github.com/PrestaShop/ps_apiresources/pull/16
| Sponsor company   | -

### CODE REVIEW

File: tests/Integration/ApiPlatform/EndPoint/ApiAccessTest.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+     */
+    public function testProtectedEndpoints(string $method, string $uri): void
+    {
+        $client = static::createClient();
+        $response = $client->request($method, $uri);
+        self::assertResponseStatusCodeSame(401);
+
+        $content = $response->getContent(false);
+        $this->assertNotEmpty($content);
+        $this->assertEquals('No Authorization header provided', $content);
```
Comment (M0rgan01) [inline tests/Integration/ApiPlatform/EndPoint/ApiAccessTest.php]: waiting for https://github.com/PrestaShop/PrestaShop/pull/35368

**Review by jolelievre — DISMISSED** [review]
Thanks @M0rgan01

File: src/PrestaShopBundle/ApiPlatform/Resources/ApiClient.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+use ApiPlatform\Metadata\ApiResource;
+use PrestaShop\PrestaShop\Core\Domain\ApiClient\Query\GetApiClientForEditing;
+use PrestaShopBundle\ApiPlatform\Metadata\CQRSGet;
+use PrestaShopBundle\ApiPlatform\Provider\QueryProvider;
+
+#[ApiResource(
+    operations: [
+        new CQRSGet(
+            uriTemplate: '/api-client/infos',
+            provider: QueryProvider::class,
```
Comment (tleon) [inline src/PrestaShopBundle/ApiPlatform/Resources/ApiClient.php]: you can remove this I think

**Review by tleon — CHANGES_REQUESTED** [review]

File: src/PrestaShopBundle/ApiPlatform/Resources/ApiClient.php (hunk 2 above)
Comment (jolelievre) [reply src/PrestaShopBundle/ApiPlatform/Resources/ApiClient.php]: a nice cacth!

**Review by jolelievre — DISMISSED** [review]

**Review by tleon — DISMISSED** [review]

**Review by jolelievre — APPROVED** [review]

**Review by tleon — APPROVED** [review]

**Pull request merged**